motionplan/motionPlanner.go

```diff
@@ -381,3 +382,81 @@ IK:
 	}
 	return orderedSolutions, nil
 }
+
+// CheckPlan checks if obstacles intersect the trajectory of the frame following the plan.
+func CheckPlan(
+	checkFrame frame.Frame,
+	plan []map[string][]frame.Input,
+	worldState *frame.WorldState,
+	fs frame.FrameSystem,
+	currentPosition spatialmath.Pose,
+	errorState spatialmath.Pose,
+) error {
+	// ensure that we can actually perform the check
+	if len(plan) < 1 {
+		return errors.New("plan must have at least one element")
+	}
+
+	// construct solverFrame
+	// Note that this requires all frames which move as part of the plan, to have an
```

Comment:
I don't understand what this means nor what variables it refers to. Do we check that the caller has guaranteed that `all frames which move as part of the plan entry in the very first plan waypoint`?  If not, what happens if the caller doesn't do that?

Reply:
what happens if the caller doesn't meet the guarantees? Does this function always succeed, always fail, or is the behavior undefined?

Reply:
Also, if the caller needs to guarantee something about the inputs, could we please document that in the comment of the function?